### PR TITLE
*: adding a code of conduct

### PR DESCRIPTION
**Pull request by vbatts**
Signed-off-by: Vincent Batts <vbatts@hashbangbash.com>

### CODE REVIEW

**Comment by wking** [comment]
This looks reasonable to me.

If you were leaning on [1] (some of the text looks similar,
e.g. “Disagreements, both social and technical, happen all the time“),
probably mention it in the commit message?  That project is CC BY 4.0
[2], and I'm not sure how that matches with the Apache 2.0 (maybe we
need to start a NOTICE file?), although I haven't compared them
closely enough to have a read on whether the borrowing is significant
enough to involve copyright concerns.

[1]: http://todogroup.org/opencodeofconduct/
[2]: https://github.com/todogroup/opencodeofconduct/blob/gh-pages/LICENSE

**Comment by vbatts** [comment]
I do link to them. Perhaps the attribution could be more clear.

IANAL, according to http://www.apache.org/legal/resolved.html#cc-sa and http://opendefinition.org/licenses/ CC-BY-4.0 and Apache-2.0 have compatibility.

**Comment by jbeda** [comment]
Some other examples crowdsourced from twitter:
https://www.kernel.org/doc/Documentation/CodeOfConflict
http://todogroup.org/opencodeofconduct/
https://docs.docker.com/opensource/governance/conduct-code/

Some of these are very long and thorough.  I like the Docker one a lot though.

**Comment by philips** [comment]
this seems reasonable and looks fine to me.

File: code-of-conduct.md [hunk 1]
```diff
@@ -0,0 +1,37 @@
...
+It is important that we resolve disagreements and differing views constructively.
+
+This code of conduct applies both within project spaces and in public spaces when an individual is representing the project or its community.
+Participants should be aware of these concerns.
+
+We are committed to making participation in this project a harassment-free experience for everyone, regardless of level of experience, gender, gender identity and expression, sexual orientation, disability, personal appearance, body size, race, ethnicity, age, religion, or nationality.
+
+Examples of unacceptable behavior by participants include:
+
+* The use of sexualized language or imagery
```
Comment (mrunalp) [inline code-of-conduct.md]: Missing full stops for the items in the list.

Comment (vbatts) [inline code-of-conduct.md]: Periods? Bullets don't get periods. Do they?

Comment (mrunalp) [inline code-of-conduct.md]: Okay, looked up grammar. The recommend that full sentences end with periods, but this list shouldn't have any. Need to remove it from the last item in the list, though.

Comment (vbatts) [inline code-of-conduct.md]: updated

**Comment by vbatts** [comment]
and added additional attribution.

**Comment by mrunalp** [comment]
LGTM

**Comment by wking** [comment]
On Fri, Sep 04, 2015 at 08:46:13AM -0700, Vincent Batts wrote:
> IANAL, according to http://www.apache.org/legal/resolved.html#cc-sa
> and http://opendefinition.org/licenses/ CC-BY-4.0 and Apache-2.0
> have compatibility.

Ah, right.  Without the SA clause your existing “Thanks” notices are
probably sufficient.

**Comment by dqminh** [comment]
LGTM

**Pull request merged**